test/e2e/test_e2e.py

```diff
@@ -0,0 +1,717 @@
+            output = fetch_data_file(fileConfig['copy_file'], fileConfig['copy_file_hash'])
+            print("Using output file {}".format(output)) 
+
+        else:
+            input = fetch_data_file(fileConfig['data_file'], fileConfig['data_file_hash'])
+            output = os.path.join(tmp_path, os.path.basename(input) + "_" + section + ".h5")
+
+            print("Convert file {} to {}".format(input, output)) 
+
+            command = ["siemens_to_ismrmrd", "-X",
+                    "-f", input, 
+                    "-m", fileConfig.get('parameter_xml', 'IsmrmrdParameterMap_Siemens.xml'),
+                    "-x", fileConfig.get('parameter_xsl', 'IsmrmrdParameterMap_Siemens.xsl'),
+                    "-o", output,
+                    "-z", str(fileConfig['measurement']),
+                    fileConfig.get('data_conversion_flag', '')]
+
+            with open(os.path.join(tmp_path, os.path.basename(input) + "_" + section + '.log.out'), 'w') as log_stdout:
+                with open(os.path.join(tmp_path, os.path.basename(input) + "_" + section + '.log.err'), 'w') as log_stderr:
+                    result = subprocess.run(command,
+                                stdout=log_stdout,
+                                stderr=log_stderr, 
+                                cwd=tmp_path)
+            
+            if result.returncode != 0:
+                pytest.fail("siemens_to_ismrmrd failed with return code {}".format(result.returncode))
+
+        return output
+    
+    return _siemens_to_ismrmrd
+
+def query_gadgetron_capabilities(info_string):
+    print("Querying Gadgetron capabilities...")
+
+    value_pattern = r"(?:\s*):(?:\s+)(?P<value>.*)?"
+
+    capability_markers = {
+        'version': "Version",
+        'build': "Git SHA1",
+        'memory': "System Memory size",
+        'python': "Python Support",
+        'julia': "Julia Support",
+        'matlab': "Matlab Support",
+        'cuda': "CUDA Support",
+        'cuda_devices': "Number of CUDA capable devices",
+    }
+
+    plural_capability_markers = {
+        'cuda_memory': "Total amount of global GPU memory"
+    }
+
+    def find_value(marker):
+        pattern = re.compile(marker + value_pattern, re.IGNORECASE)
+        match = pattern.search(info_string)
+
+        if not match:
+            raise RuntimeError("Failed to parse Gadgetron info string; Gadgetron capabilities could not be determined.")
+
+        return match['value']
+
+    def find_plural_values(marker):
+        pattern = re.compile(marker + value_pattern, re.IGNORECASE)
+        return [match['value'] for match in pattern.finditer(info_string)]
+
+    capabilities = {key: find_value(marker) for key, marker in capability_markers.items()}
+    capabilities.update({key: find_plural_values(marker) for key, marker in plural_capability_markers.items()})
+
+    return capabilities
+
+@pytest.fixture
+def check_requirements(host_url, port, external, ignore_requirements, run_tag):
```

Comment:
This capability parsing stuff is convoluted to the point of being opaque. I realize some of it is inherited, but let's clean and improve and not make worse. I would suggest a (data) class that defines (with default values) cabalilities that a Gadgetron instance can have and then populate that with a simple parsing process here.